src/rust/engine/src/nodes.rs

```diff
@@ -735,9 +786,9 @@ impl DownloadedFile {
 
 #[async_trait]
 impl WrappedNode for DownloadedFile {
-  type Item = Arc<store::Snapshot>;
+  type Item = Digest;
```

Comment:
I would have done this change back in https://github.com/pantsbuild/pants/pull/10449, I only didn't know the change was possible.